lib/compress.js

```diff
@@ -4307,6 +4307,17 @@ merge(Compressor.prototype, {
         return self;
     });
 
+    AST_Lambda.DEFMETHOD("contains_this", function() {
+        var result;
+        var self = this;
+        self.walk(new TreeWalker(function(node) {
+            if (result) return true;
+            if (node instanceof AST_This) return result = true;
+            if (node !== self && node instanceof AST_Scope) return true;
+        }));
+        return result;
+    });
```

Comment:
This needs to merge with https://github.com/mishoo/UglifyJS2/blob/f5c46db7386870032d6d0266b2fe97686dfb7715/lib/compress.js#L4659-L4668